lcls_tools/common/data_analysis/projection_fit/method_base.py

```diff
@@ -0,0 +1,105 @@
+from abc import ABC, abstractmethod
+import numpy as np
+from matplotlib import pyplot as plt
+
+
+class MethodBase(ABC):
+    """
+    Base abstract class for all methods, which serves as the bare minimum skeleton code needed.
+    Should be used only as a parent class to all method models.
+    ---------------------------
+    Arguments:
+    param_names: list (list all of param names that the model will contain)
+    param_guesses: np.ndarray (array that contains a guess as to what
+        each param value is organized in the same order as param_names)
+    param_bounds: np.ndarray (array that contains the lower
+        and upper bound on for acceptable values of each parameter)
+    """
+
+    def __init__(self):
+        self.param_names: list = None     
+        self.param_bounds: np.ndarray = None
+        self.init_values: dict = None
+        self.init_values_list: list = None
```

Comment:
Recall that projection_fit.py needs a list in order to use fit_model. If you change the type of self.init_values to a dict and unpack it to an ordered list inside fit model then now ProjectionFit becomes model dependent instead of a template. (knowledge or gaussian model params should not be in projection_fit). Recommending instead make a dictionary and a list of param values.

Reply:
see my other comment on why init_values_list is not needed